File: src/c++/graph/bfs/5567/main.cc
/* Date : 2019.5.10
 * Author : seungyeon choi
 * 5567 (결혼식)
 */

#include <cmath>
#include <cstdio>
#include <queue>

using namespace std;

const int MAX_FRIEND = 501;

int N, M;
int adj[MAX_FRIEND][MAX_FRIEND];
bool visit[MAX_FRIEND];

int bfs() {
  int x, f, ans = 0;
  queue<pair<int, int>> q;
  q.push(make_pair(x, f));

  while (!q.empty()) {
    x = q.front().first;
    f = q.front().second;
    q.pop();
    visit[x] = true;

    for (int i = 0; i < N; i++) {
      if (adj[x][i]) {
        if (!visit[i]) {
          visit[i] = true;
          q.push(make_pair(i, f + 1));
          if (f + 1 <= 2)
            ans += 1;
        }
      }
    }
  }

  return ans;
}

int main() {
  int num1, num2;
  scanf("%d", &N);
  scanf("%d", &M);
  for (int i = 0; i < M; i++) {
    scanf("%d %d", &num1, &num2);
    adj[num1 - 1][num2 - 1] = 1;
    adj[num2 - 1][num1 - 1] = 1;
  }
  printf("%d\n", bfs());
  return 0;
}

File: src/python/graph/bfs/5567/main.py
# Date : 2017.11.15
# Author : seungyeon choi
# 5567 (결혼식)

N = int(input())
M = int(input())
adj = [[] for _ in range(N)]
for _ in range(M):
    A, B = map(int, input().split())
    adj[A-1].append(B-1)
    adj[B-1].append(A-1)
ans = 0
q = []
visit = [0 for _ in range(N)]
q.append((0, 0))
while q :
    x, w = q.pop(0)
    visit[x] = 1
    for j in range(len(adj[x])):
        temp = adj[x][j]
        if not visit[temp] :
            visit[temp] = 1
            q.append((temp, w+1))
            if w+1 <= 2:
                ans += 1
print (ans)
